service/src/main/java/bio/terra/tanagra/service/accesscontrol/impl/AouWorkbenchAccessControl.java

```diff
@@ -109,31 +115,10 @@ public boolean isAuthorized(UserId user, Permissions permissions, @Nullable Reso
   @Override
   public ResourceCollection listAllPermissions(
       UserId user, ResourceType type, @Nullable ResourceId parentResource, int offset, int limit) {
-    if (ResourceType.ACTIVITY_LOG.equals(type)) {
-      LOGGER.warn("AoU-RW will not use the activity log, so we should never get here.");
-      return ResourceCollection.empty(ResourceType.ACTIVITY_LOG, null);
-    } else if (ResourceType.UNDERLAY.equals(type)) {
-      // Browsing the cohort builder without saving. Allow for all underlays.
-      return ResourceCollection.allResourcesAllPermissions(ResourceType.UNDERLAY, null);
-    } else if (ResourceType.STUDY.equals(type)) {
-      // Call the workbench to get the list of workspaces the user can see, and their role on each.
-      Map<String, WorkspaceRole> workspaceRoleMap = apiListWorkspacesWithAccess(user);
-
-      // Map the workspace roles to Tanagra permissions.
-      Map<ResourceId, Permissions> studyPermissionsMap = new HashMap<>();
-      workspaceRoleMap.entrySet().stream()
-          .forEach(
-              entry -> {
-                String studyId = entry.getKey();
-                WorkspaceRole role = entry.getValue();
-                studyPermissionsMap.put(
-                    ResourceId.forStudy(studyId),
-                    Permissions.forActions(
-                        ResourceType.STUDY, role.getTanagraPermissions().get(ResourceType.STUDY)));
-              });
-      return studyPermissionsMap.isEmpty()
-          ? ResourceCollection.empty(type, parentResource)
-          : ResourceCollection.resourcesDifferentPermissions(studyPermissionsMap);
+    if (ResourceType.UNDERLAY.equals(type)) {
```

Comment:
Per comment below, should this be `UNDERLAY || STUDY`?

Reply:
Just UNDERLAY, because  the if logic in else will take care of STUDY, since parentResource for STUDY would be null for listAllPermissions call
```
     if (parentResource == null || parentResource.getStudy() == null) {
        LOGGER.error(
            "Study id is required to list permissions for a child artifact: {}", parentResource);
        return ResourceCollection.empty(type, parentResource);
      }

```

Reply:
Ah OK -- in that case, can you update the log message there please? i.e. we should never get here ... as the underlays one above.

Reply:
I think it is better to use 
```
if (ResourceType.UNDERLAY.equals(type) || ResourceType.STUDY.equals(type)) {
      // AoU will not call to list Underlays or list studies - this is managed by Workbench
      LOGGER.error("Calls from AoU Workbench should never get here. ");
      return ResourceCollection.empty(type, null); 
```
where it is explici that AoU should never call list underlays or list studies.